CircularBuffer.h

```diff
@@ -102,27 +119,55 @@ template<typename T, size_t S, typename IT = typename Helper::Index<(S <= UINT8_
 	 */
 	T shift();
 
+	/**
+	 * @brief Removes an element from the beginning of the buffer.
+	 *
+	 * @warning Calling this operation on an empty buffer has an unpredictable behaviour.
+	 */
+	void shift(T &);
```

Comment:
how is a user protected against using the wrong version of the shift function if they are both available?

Reply:
There is no wrong version of shift(), just user preference.